templates/software_router/k8s-software-router.yaml

```diff
@@ -36,6 +36,13 @@ spec:
         - name: {{ .Chart.Name }}-k8s-software-router
           image: "{{ .Values.k8s_software_router.image.repository }}:{{ .Values.k8s_software_router.image.tag }}"
           imagePullPolicy: {{ .Values.k8s_software_router.image.pullPolicy }}
+          livenessProbe:
+            exec:
+              command:
+                - pgrep
+                - python3
+            initialDelaySeconds: 10
+            timeoutSeconds: 10
```

Comment:
Question: Why not add readiness probe to this deployment?

Reply:
Mostly because I felt so silly doing the liveness probe :) (The way it is done, the container will die if the python process goes away.)

It's not wrong to have a `readinessProbe`, but adds more value in cases where something "dials in" to the container (so there's a service that must be available) and even more so if it takes 10 minute for it to start.

(Another case is when restarting statefulsets or e.g. replacing deployments where kubernetes will wait until a pod is ready before continuing, but here we can't really do any meaningful test.)

Reply:
Understood. Yeah, it makes sense. It's fine to skip it. :+1:

Reply:
Understood. Yeah, it makes sense. It's fine to skip it. :+1:

Reply:
Understood. Yeah, it makes sense. It's fine to skip it. :+1: